platform.md

```diff
@@ -133,6 +122,410 @@ A platform MAY support any number of mixins for a given stack in order to suppor
 Changes introduced by mixins SHOULD be restricted to the addition of operating system software packages that are regularly patched with strictly backwards-compatible security fixes.
 However, mixins MAY consist of any changes that follow the [Compatibility Guarantees](#compatibility-guarantees).
 
+## Lifecycle Interface
+### Platform API Compatibility
+
+The platform SHOULD set `CNB_PLATFORM_API=<major>[.<minor]` is set in the lifecycle's execution environment
+
+IF `CNB_PLATFORM_API=<major>[.<minor]` is set in the lifecycle's execution environment, the lifecycle MUST do one the following before attempting to parse other inputs
+  1. conform usage and behavior to the given version of the platform API specification
+  1. fail if it does not support the requested API version
+
+### Operations
+
+#### Build
+A single app image build consists of the following phases:
+
+1. Detection
+1. Analysis
+1. Cache Restoration
+1. Build
+1. Export
+
+A platform MUST execute these phases EITHER by invoking the following phase-specific lifecycle binaries in order:
+1. `/cnb/lifecycle/detector`
+1. `/cnb/lifecycle/analyzer`
+1. `/cnb/lifecycle/restorer`
+1. `/cnb/lifecycle/builder`
+1. `/cnb/lifecycle/exporter`
+
+OR by executing `/cnb/lifecycle/creator`.
+
+#### Rebase
+Run image rebasing allows for fast stack updates for already-exported OCI images without require a rebuild. A rebase requires minimal data transfer when those images are stored on a Docker registry.
+When a new stack version with the same stack ID is available, the app layers and launch layers SHOULD be rebased on the new run image by updating the image's configuration to point at the new run image.
```

Comment:
Do we need to be more explicit than "rebased on the new run image"? Maybe something like:

> create a new image using the app layers and launch layer on the new run image ...

Also, I'm not great at spec-speak, but maybe this should be "MUST" instead of "SHOULD"?

Reply:
> Do we need to be more explicit than "rebased on the new run image"?

I took another stab at it. Let me know what you think.

> Also, I'm not great at spec-speak, but maybe this should be "MUST" instead of "SHOULD"?

I think this is properly a `SHOULD`. It's up to the platform to kick off a rebase when a new run image is available. Platforms like `pack` can't do this automatically every time a new stack image is released. Therefore it is guidance to platform and/or end users rather than a requirement.